### ISSUE TITLE
Setup CI pipeline

### ISSUE THREAD

**Issue by dabbertorres** [SFML/SFML.Net#151]
> I have nothing against keeping it official on this repository. The thing is, I am unable to review technical stuff, so I have to blindly rely on you guys. So I can merge stuff and maintain this repository, but you all have to agree and give me a clear "ok let's merge it" for things to move on. As long as PRs are being discussed, or no feedback is given, I can't do anything.
https://github.com/SFML/SFML.Net/pull/143#issuecomment-367941283

One way to solve this would be to setup a CI config (I can do that, ie: travis-ci.org), and configure it to require some number of approved reviews before merging is allowed - that way the process can be somewhat more automated, and community driven, more-or-less.

Plus, we can get automated building, and (if they get written) automated testing, which makes approval even easier.

Thoughts? @LaurentGomila

tl;dr: If we setup an automated pipeline of sorts, can we help things along and absolve @LaurentGomila of work he doesn't have time for?

**Comment by BenjaminHolland** [comment]
I'm a bit surprised this hasn't been done already. @dabbertorres have you done any work on this sincethe issue was opened?

**Comment by eXpl0it3r** [comment]
Anyone willing to chip in here with getting some CI up and running?

**Comment by DemoXinMC** [comment]
I am currently learning Travis CI for this exact purpose.

**Comment by DemoXinMC** [comment]
At a most basic level, I feel this could be done with PRs being checked for building without errors, and then merged PRs to the master branch triggering CI releases?  I know there was some discussion regarding whether it's advisable to have a new release for each PR or not, so I'd like some more opinions here on exactly what format we'd want.

**Comment by eXpl0it3r** [comment]
- Build PRs
- Build master branch
- Publish artifacts for git-tagged releases (depending on how what the built binaries actually look like)

**Comment by DemoXinMC** [comment]
@eXpl0it3r This issue should be able to be closed now, as CI is available.

**Comment by eXpl0it3r** [comment]
Implemented with #170
